docs/reference/ui/components.md

```diff
@@ -0,0 +1,182 @@
+(ui_components)=
+
+# UI components
+
+This document provides a reference for Wagtail's user interface (UI) components, which are used to build the features within the CMS. A demonstration of these components are available through the [Styleguide app](styleguide) and the [pattern library](pattern_library).
+
+````{note}
+```{include} ../../../client/README.md
+:start-after: <!-- STABILITY:START -->
+:end-before: <!-- STABILITY:END -->
+```
+````
+
+## Fragment
+
+```{eval-rst}
+.. autofunction:: wagtail.admin.templatetags.wagtailadmin_tags.fragment
+```
+
+## Dialog
+
+```{eval-rst}
+.. class:: wagtail.admin.templatetags.wagtailadmin_tags.DialogNode
+   :no-contents-entry:
```

Comment:
Add the corresponding `class` so that we can refer to it (and can autolink to the source code once we've set up the [linkcode](https://www.sphinx-doc.org/en/master/usage/extensions/linkcode.html) extension).

`:no-contents-entry:` so that we don't add an unnecessary entry to the TOC.